### PR TITLE
feat(NODE-5838): remove support for server 3.6

### PR DESCRIPTION
**Pull request by durran**
### Description

Removes support for MongoDB 3.6

#### What is changing?
- Syncs SDAM spec tests and updates the incorrect ones where maxWireVersion was set to 6.
- Updates the incorrect max staleness spec tests with the same issue.
- Removes MONGODB-CR auth.
- Removes 3.6 server from the Evergreen CI matrix.

##### Is there new documentation needed for these changes?

Need to communicate one minor release prior.

#### What is the motivation for this change?

NODE-5838

### MongoDB 3.6 is no longer supported

Server versions 3.6 and lower will get a compatibility error on connection and support for MONGODB-CR authentication is now also removed.

### Double check the following

- [x] Ran `npm run check:lint` script
- [x] Self-review completed using the [steps outlined here](https://github.com/mongodb/node-mongodb-native/blob/HEAD/CONTRIBUTING.md#reviewer-guidelines)
- [x] PR title follows the [correct format](https://www.conventionalcommits.org/en/v1.0.0/): `type(NODE-xxxx)[!]: description`
  - Example: `feat(NODE-1234)!: rewriting everything in coffeescript`
- [x] Changes are covered by tests
- [ ] New TODOs have a related JIRA ticket

### CODE REVIEW

File: .evergreen/ci_matrix_constants.js
Comment (aditi-khare-mongoDB) [inline .evergreen/ci_matrix_constants.js:1]: Some API docs still refer to 3.6:
- [this line](https://github.com/mongodb/node-mongodb-native/blob/8b0f3541a75a7145a36ffd9f543dc3665e1701a3/src/db.ts#L282)  in `db.ts`
- [these lines ](https://github.com/mongodb/node-mongodb-native/blob/8b0f3541a75a7145a36ffd9f543dc3665e1701a3/src/explain.ts#L14-L17) in `explain.ts`.

Can we remove these comments?

File: test/unit/connection_string.spec.test.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@ import { executeUriValidationTest } from '../tools/uri_spec_runner';
 
 const skipTests = [
   // TODO(NODE-3914): Fix; note that wtimeoutms will be deprecated via DRIVERS-555 (NODE-3078)
-  'Deprecated (or unknown) options are ignored if replacement exists'
+  'Deprecated (or unknown) options are ignored if replacement exists',
+  'Colon in a key value pair'
```
Comment (aditi-khare-mongoDB) [inline test/unit/connection_string.spec.test.ts:7]: Can we create a ticket with the `spec-compliance` label about the failing 'Colon in a key value pair' test?

**Review by aditi-khare-mongoDB — CHANGES_REQUESTED** [review]

File: test/unit/connection_string.spec.test.ts (hunk 1 above)
Comment (durran) [reply test/unit/connection_string.spec.test.ts:7]: There's no ticket as we are not going to fix that. Node's URI class cannot handle it and it was deemed acceptable that this specific case is provided as a MongoClient option only. See the TOKEN_RESOURCE section in MONGODC-OIDC auth mech propeties: https://github.com/mongodb/specifications/blob/master/source/auth/auth.md#machine-authentication-flow

File: .evergreen/ci_matrix_constants.js
Comment (durran) [reply .evergreen/ci_matrix_constants.js:1]: Comments removed.

File: test/unit/connection_string.spec.test.ts (hunk 1 above)
Comment (durran) [reply test/unit/connection_string.spec.test.ts:8]: Add added a comment to the test to note this.

**Review by aditi-khare-mongoDB — APPROVED** [review]

**Review by baileympearson — APPROVED** [review]

**Pull request merged**